docs/frontend-system/building-plugins/01-index.md

```diff
@@ -0,0 +1,203 @@
+
+To create a frontend plugin, run `yarn new`, select `plugin`, and fill out the rest of the prompts. This will create a new package at `plugins/<pluginId>`, which will be the main entrypoint for your plugin.
+
+> **NOT: The created plugin will currently be templated for use in the legacy frontend system, and you will need to replace the existing plugin wiring code.**
+
+## The plugin instance
+
+The starting point of a frontend plugin is the `createPlugin` function, which accepts a single options object as its only parameter. It is imported from `@backstage/frontend-plugin-api`, which is where you will find most of the common APIs for building plugins.
+
+This is how to create a minimal plugin:
+
+```tsx title="in src/plugin.ts"
+import { createPlugin } from '@backstage/frontend-plugin-api';
+
+export const examplePlugin = createPlugin({
+  id: 'example',
+  extensions: [],
+});
+```
+
+```tsx title="in src/index.ts"
+export { examplePlugin as default } from './plugin';
+```
+
+Note that we export the plugin as the default export of our package from `src/index.ts`. This is important, as it is how users of our plugin are able to seamlessly install the plugin package in a Backstage app without having to reference the plugin instance through code.
+
+The plugin ID should be a lowercase dash-separated string, while the plugin instance variable should be the camel case version of the ID with a `Plugin` suffix. For more details on naming patterns within the frontend system, see [the article on naming patterns](../architecture/08-naming-patterns.md). By sticking to these naming patterns you ensure that users of your plugin more easily recognize the exports and features provided by your plugin.
+
+## Adding extensions
+
+The plugin that we created above is empty, and doesn't provide any actual functionality. To add functionality to a plugin you need to create and provide it with one or more [extensions](../architecture/03-extensions.md). Let's continue by adding a standalone page to our plugin, as well as a navigation item that allows users to navigate to the page.
+
+To create a new extension you typically use pre-defined [extension creators](../architecture/03-extensions.md#extension-creators), provided either by the framework itself or by other plugins. In this case we'll need to use `createPageExtension` and `createNavItemExtension`, both from `@backstage/frontend-plugin-api`. We will also need to [create a route reference](../architecture/07-routes.md#creating-a-route-reference) to use as a reference for out page, allowing us to dynamically create URLs that link to our page.
```

Comment:
Yeah it's a fairly common error. It's kinda tricky to move them together more tbh, we need the separation of the route refs to avoid circular imports, and the plugin instance isn't able to keep into the extension factories to see what route refs they will provide. Might be an opportunity for lint rules?

Reply:
That sounds like a good idea, could you please add it to the Frontend system meta issue?